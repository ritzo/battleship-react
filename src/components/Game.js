import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import BoardContainer from '../containers/BoardEditorContainer';

import './Game.css';

class Game extends React.Component {
  constructor(props) {
    super(props);

    this.gameData = props.gameData;
    this.actions = props.actions;
  }

  render() {
    return (
      <div>
        <h1>Game</h1>

        <span>Hello</span>

        <b>{this.gameData.player}</b>
        <br />
        <BoardContainer />

        <Link to="/gameOver">Surrender</Link>
      </div>
    );
  }
}

Game.propTypes = {
  gameData: PropTypes.shape({
    player: PropTypes.string,
    playerBoard: PropTypes.array.isRequired,
    playerBoardLast: PropTypes.number,
    oponentBoard: PropTypes.array.isRequired,
    oponentBoardLast: PropTypes.number,
    state: PropTypes.string.isRequired,
    next: PropTypes.string,
    winner: PropTypes.string,
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  actions: PropTypes.object.isRequired,
};

export default Game;
