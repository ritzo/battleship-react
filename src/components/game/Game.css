@media only screen and (min-width: 801px) {
    .board-container {
        width: 45%;
        display: inline-block;
        padding-left: 2%;
    }
}

@media only screen and (max-width: 800px) {
    .board-container {
        width: 100%;
        display: inline-block;
        padding-left: 2%;
    }
}


.next-turn {
    font-weight: bold;
}

.surrender {
    float: right;
    margin-right: 50px;
    padding: 10px;
}
