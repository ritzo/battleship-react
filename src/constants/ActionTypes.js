// Actions related to the game setup
export const SETUP_BOARD = 'SETUP_BOARD';
export const SET_NAME = 'SET_NAME';
export const ADD_SHIP = 'ADD_SHIP';
export const PLAY = 'PLAY';
// Actions related to the game itself
export const SURRENDER = 'SURRENDER';
export const SHOOT = 'SHOOT';
export const OPONENT_TURN = 'OPONENT_TURN';
