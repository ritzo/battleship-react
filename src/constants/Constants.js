export const STATES = {
  OPEN: 'open',
  PLAYING: 'playing',
  FINISHED: 'finished',
};

export const SHIPS = [2, 3, 3, 3, 4];

export const POSITIONS = {
  HORIZONTAL: 'horizontal',
  VERTICAL: 'vertical',
};

export const PLAYERS = {
  PLAYER: 'player',
  OPONENT: 'cpu',
};

export const SQUARE_STATES = {
  EMPTY: 'empty',
  INVALID: 'invalid',
  SHIP: 'ship',
  SHOOT_MISS: 'shoot_miss',
  SHOOT_SHIP: 'shoot_ship',
};

export const BOARD_TYPES = {
  EDITION: 'edition',
  PLAYER: 'player',
  OPONENT: 'oponent',
};
