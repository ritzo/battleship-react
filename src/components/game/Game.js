import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';

import BoardPlayerContainer from '../board/BoardPlayerContainer';
import BoardOponentContainer from '../board/BoardOponentContainer';

import { BOARD_TYPES, STATES } from '../../constants/Constants';

import './Game.css';

class Game extends React.Component {
  constructor(props) {
    super(props);

    this.actions = props.actions;
  }

  handleSurrenderClick() {
    this.actions.surrender();
  }

  render() {
    const {
      player,
      next,
      state,
    } = this.props;

    /*
    if (state === STATES.OPEN) {
      return <Redirect to="/" />;
    }
    */

    if (state === STATES.FINISHED) {
      return <Redirect to="/gameOver" />;
    }

    return (
      <div>
        <h1>Game</h1>

        <span>Hello</span>

        <b>{ player }</b>
        <br />

        <BoardPlayerContainer
          title="Your fleet"
          type={BOARD_TYPES.PLAYER}
        />

        <button type="button" onClick={() => this.handleSurrenderClick()}>Surrender</button>
      </div>
    );
  }
}

Game.propTypes = {
  player: PropTypes.string.isRequired,
  next: PropTypes.string.isRequired,
  state: PropTypes.string.isRequired,
  actions: PropTypes.shape({
    surrender: PropTypes.func.isRequired,
  }).isRequired,
};

export default Game;
