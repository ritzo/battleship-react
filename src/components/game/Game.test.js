import React from 'react';
import { shallow } from 'enzyme';

import Game from './Game';

import { PLAYERS, STATES } from '../../constants/Constants';

describe('Game', () => {
  it('Snapshot', () => {
    const player = 'someone';
    const next = PLAYERS.PLAYER;
    const state = STATES.PLAYING;
    const actions = {
      surrender() {},
    };

    const enzymeWrapper = shallow(
      <Game player={player} next={next} state={state} actions={actions} />,
    );

    expect(enzymeWrapper).toMatchSnapshot();
  });
});
