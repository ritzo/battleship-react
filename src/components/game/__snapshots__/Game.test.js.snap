// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Snapshot 1`] = `
<div>
  <div>
    <div
      className="board-container"
    >
      <Connect(Board)
        title="someone's fleet"
        type="player"
      />
    </div>
    <div
      className="board-container"
    >
      <Connect(Board)
        title="CPU's fleet"
        type="oponent"
      />
    </div>
  </div>
  <p
    className="next-turn"
  >
    someone's turn
  </p>
  <button
    className="surrender"
    onClick={[Function]}
    type="button"
  >
    Surrender
  </button>
</div>
`;
